spec/jobs/convert_document_job_spec.rb

```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe ConvertDocumentJob do
+  subject { described_class.new }
+
+  let(:collection) { create(:collection) }
+  let(:user) { create(:user) }
+  let(:file) { nil }
+  let(:filename) { nil }
+  let(:link) { nil }
+  let(:chunking_method) { :basic }
+  let(:chunking_profile) { create(:chunking_profile, method: chunking_method, size: 800) }
+  let(:doc) { create(:document, state: :created, link:, file:, filename:, chunking_profile:, collection:, user:) }
+
+  describe "#perform" do
+    context "with a PDF document" do
+      let(:filename) { "spec/fixtures/files/gnu_manifesto.pdf" }
+      let(:file) { fixture_file_upload("gnu_manifesto.pdf") }
+
+      before do
+        allow(Mediator).to receive("ingest")
+      end
+
+      it "calls Mediator after conversion" do
+        subject.perform(doc.id)
+
+        expect(Mediator).to have_received("ingest")
```

Comment:
That's  lot of setup for just this example.

Should you also check that `ingest` got the right arg, using `.with(doc)` or something like that?

Reply:
Looks like the code should also:

- update the doc after the conversion
- not call `Mediator#ingest` if the doc is `errored?`

Should those be tested as well?

Reply:
This is related to the earlier comment about `ConvertDocumentJob` exposing `new_doc` attribute. I would need to re-expose that to use `.with(...)` in this test. At this point, all I care about is that the conversion sends a new doc to `Mediator`. The converters are themselves tested on their own.

Reply:
You can use `Document.last` instead of `new_doc`, and add an example that checks that `Document.count` went up by 1, so you know you have the right doc.